propolis/src/hw/virtio/viona.rs

```diff
@@ -450,6 +452,32 @@ impl VionaPoller {
         }
     }
 }
+
+// macOS doesn't expose the epoll_create1 function as well as some other
+// constants used above. Given viona isn't available on non-illumos systems
+// anyways, we stub with just enough that it builds and can run unit tests.
+#[cfg(not(target_os = "illumos"))]
+impl VionaPoller {
+    fn spawn(
+        viona_fd: RawFd,
+        dev: Weak<VirtioViona>,
+        ctx: &DispCtx,
+    ) -> Result<(Arc<Self>, AsyncTaskId)> {
+        let this = Arc::new(Self { epfd: viona_fd, dev });
+        let for_spawn = Arc::clone(&this);
+        let task = ctx.spawn_async(move |actx| async move {
+            let _ = for_spawn.poll_interrupts(&actx).await;
+        });
+        Ok((this, task))
```

Comment:
Since there's no way this'll work on macos anyways, why not simply toss an error like `event_present`?